pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -326,14 +288,13 @@ func (p *PriorityQueue) Add(pod *v1.Pod) error {
 	}
 	if p.unschedulableQ.get(pod) != nil {
 		klog.Errorf("Error: pod %v/%v is already in the unschedulable queue.", pod.Namespace, pod.Name)
-		p.deleteNominatedPodIfExists(pod)
```

Comment:
Souldn’t we make sure that the pod in the unschedulable queue isn’t nominated anymore by calling a delete ?

Reply:
Agree. Miss `p.nominatedPods.delete(pod)` here?

Reply:
ok, I think it's because at Line 297, `p.nominatedPods.add(pod, "")` implicitly delete the pod from cache. But it sounds weird to me that we delete pod in an "add" function?

Reply:
As @Huang-Wei noted, this is no longer needed as the `add` function of nominated node struct ensures that there won't be duplicate entries. I think `add` is still the best name for it. Any other suggestions?